pyproject.toml

```diff
@@ -13,6 +13,7 @@ packages = [
 [tool.poetry.dependencies]
 python = "^3.9"
 arcaflow-plugin-sdk = "^0.13.0"
+requests = "^2.31.0"
```

Comment:
This may have interesting implications. You're switching from (presumably) an RPM `requests` (as `requests` wasn't apparently listed before as a Python dependency) to a PyPi `requests`. When I accidentally switched an environment from the Fedora build of `requests` (which uses the `/etc/pki/tls/certs/ca-bundle.crt` bundle, which, if installed, includes the Red Hat IT CA) to the PyPi build of the identically versioned `requests` (which uses `certifi`, a Google CA bundle which does *not* include Red Hat IT CA), I was unable to verify TLS connections to internal HTTPS services (like the RDU2 Pbench server).

I'm not entirely sure what this means for CentOS stream9, or for requests 2.31 builds (i.e., is there a stream9 RPM?), or even whether your use case requires internal TLS; but the fact that "requests 2.31" may not be the same as "requests 2.31" is something to keep in mind...

Reply:
IIRC, the failure mode was even more subtle than what you're relaying (I hit this, too!).  I don't recall the details well, but one of the versions of `requests` would use use `certifi` _only if it was installed_, and it would try its best to work without it if it was missing...which didn't work well if you wanted to use a custom CA bundle.

Reply:
Your "subtlety" was on the PyPi module side, whereas I was focusing on the other side. My point is that the `certifi` version (regardless of whether `certifi` itself is installed) doesn't use the Fedora standard `/etc/pki/tls/certs/ca-bundle.crt` unless you define `REQUESTS_CA_BUNDLE=/etc/pki/tls/certs/ca-bundle.crt`. The Fedora RPM version does so implicitly.

Reply:
I'd have to dig a bit, but I think the dependency of `requests` came in new with the updated PCP version. Or it's possible that the RPM was in the base build of stream8 and is not in stream9. Either way, I don't think this introduces a direct maintenance conflict. I could choose to either deploy this via pypi or via RPM. :shrug:

Reply:
Checking the existing plugin based on stream8, and I don't see `requests` installed via either rpm or pypi.

Reply:
Interesting. As long as you're not using TLS connections (or only using CA certs from the Google `certifi` bundle), which `requests` you're using probably doesn't matter: it's just one more gem from configuration management purgatory ...